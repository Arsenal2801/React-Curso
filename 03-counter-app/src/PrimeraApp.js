
import React from 'react';
import PropTypes from 'prop-types';


const PrimeraApp = ({ saludo, subtitulo }) => {

    // const saludo = {
    //     nombre: 'German',
    //     edad: 32,
    // };




    return (
        <>
            <h1>Hola Mundo</h1>
            <h4>Mi primera App</h4>
            <h5>{saludo}</h5>
            <p>{subtitulo}</p>
            {/* <pre>{JSON.stringify(saludo)}</pre> */}
        </>
    );

};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}


export default PrimeraApp;

