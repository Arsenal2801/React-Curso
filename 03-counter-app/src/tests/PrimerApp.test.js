import PrimeraApp from "../PrimeraApp";
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

describe('Pruebas de Componentes(PrimeraApp)', () => {
    // test('Muestra Hola, Soy Goku', () => { 
    //     const saludo = 'Hola, Soy Goku';
    //     const {getByText} = render(<PrimeraApp saludo={saludo} />)
    //     expect(getByText(saludo)).toBeInTheDocument();
    //  })


    test('Muestra <PrimeraApp/> correctamente', () => {
        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)

        expect(wrapper).toMatchSnapshot();

    })

    test('Muestra en <PrimeraApp/> el subtitulo ', () => {

        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Hola, Soy Vegeta';
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        )

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);

    })



})

