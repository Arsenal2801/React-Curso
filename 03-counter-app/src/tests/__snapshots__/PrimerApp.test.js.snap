// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Componentes(PrimeraApp) Muestra <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <h4>
    Mi primera App
  </h4>
  <h5>
    Hola, Soy Goku
  </h5>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas de Componentes(PrimeraApp) Muestra en <PrimeraApp/> el subtitulo  1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <h4>
    Mi primera App
  </h4>
  <p>
    Hola, Soy Goku
  </p>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
